src/fenics_constitutive/models/spring_maxwell_model.py

```diff
@@ -31,184 +34,61 @@ def __init__(
-
-            case StressStrainConstraint.PLANE_STRESS:
-                self.D_0 = (
-                    self.E0
-                    / (1 - self.nu**2.0)
-                    * np.array(
-                        [
-                            [1.0, self.nu, 0.0, 0.0],
-                            [self.nu, 1.0, 0.0, 0.0],
-                            [0.0, 0.0, 0.0, 0.0],
-                            [0.0, 0.0, 0.0, (1.0 - self.nu)],
-                        ]
-                    )
-                )
-                self.D_1 = (
-                    self.E1
-                    / (1 - self.nu**2.0)
-                    * np.array(
-                        [
-                            [1.0, self.nu, 0.0, 0.0],
-                            [self.nu, 1.0, 0.0, 0.0],
-                            [0.0, 0.0, 0.0, 0.0],
-                            [0.0, 0.0, 0.0, (1.0 - self.nu)],
-                        ]
-                    )
-                )
-
-            case StressStrainConstraint.UNIAXIAL_STRESS:
-                self.D_0 = np.array([[self.E0]])
-                self.D_1 = np.array([[self.E1]])
-            case _:
-                msg = "Constraint not implemented"
-                raise NotImplementedError(msg)
+        law = get_elasticity_law(constraint)
+        self.D_0 = law.get_D(self.E0, self.nu)
+        self.D_1 = law.get_D(self.E1, self.nu)
 
     def evaluate(
         self,
-        time: float,
+        t: float,
         del_t: float,
         grad_del_u: np.ndarray,
-        mandel_stress: np.ndarray,
+        stress: np.ndarray,
         tangent: np.ndarray,
         history: np.ndarray | dict[str, np.ndarray] | None,
     ) -> None:
+        _ = t
         assert (
             grad_del_u.size // (self.geometric_dim**2)
-            == mandel_stress.size // self.stress_strain_dim
+            == stress.size // self.stress_strain_dim
             == tangent.size // (self.stress_strain_dim**2)
         )
 
         # reshape gauss point arrays
         n_gauss = grad_del_u.size // (self.geometric_dim**2)
-        mandel_view = mandel_stress.reshape(-1, self.stress_strain_dim)
+        mandel_view = stress.reshape(-1, self.stress_strain_dim)
 
         strain_increment = strain_from_grad_u(grad_del_u, self.constraint).reshape(
             -1, self.stress_strain_dim
         )
+        if history is None:
+            msg = "history must not be None"
+            raise ValueError(msg)
         strain_visco_n = history["strain_visco"].reshape(-1, self.stress_strain_dim)
         strain_n = history["strain"].reshape(-1, self.stress_strain_dim)
 
-        # if del_t == 0:
-        #     # linear step visko strain is zero
-        #     D = self.D_0 + self.D_1
-        #     mandel_view += strain_increment @ D
-        #     _deps_visko = np.zeros_like(strain_increment)
-        # else:
         assert del_t > 0, "Time step must be defined and positive."
 
+        mu1, _ = lame_parameters(self.E1, self.nu)
```

Comment:
Why are we computing mu1 in each evaluation step, though D0 and D1 are both precomputed. I think each of the approaches has some pros and cons, though I personally see no reason to follow a different approach for both.